src/index.js

```diff
@@ -0,0 +1,165 @@
+* backend also supports domain and a secure flag.  You can also
+* set filters for the key's and values.  These can be used
+* to modify things like json encoding objects for storage,
+* or adding a prefix to the storage key.
+*
+* NOTE: The storage backends need to be registerd in the order
+* in which they should be tried.  So typically that would be this:
+*
+* - localStorage
+* - cookie
+* - memory
+*
+* @constructor Storage
+* @param {object} [opts] - The storage options
+*/
+var Storage = module.exports = function (opts = {}) {
+	// Ensure it is created with new
+	if (!(this instanceof Storage)) {
+		return new Storage(opts);
+	}
+
+	// Merge options
+	this.options = Object.assign({}, Storage.defaultOptions, opts);
+
+	// Get the backend
+	var Backend;
+	if (typeof this.options.backend === 'string') {
+		Backend = getBackend(this.options.backend);
+	} else if (this.options.backend) {
+		Backend = this.options.backend;
+	} else {
+		Backend = selectBackend();
+	}
+
+	// Verify that we got a backend
+	if (!Backend) {
+		return;
+	}
+
+	// Create backend instance
+	this.backend = new Backend();
+};
+
+/**
+ * Default options
+ *
+ * @var defaultOptions
+ * @memberof Storage
+ * @static
+ */
+Storage.defaultOptions = {
+	backend: null,
+	domain: null,
+	setValueFilter: ident,
+	getValueFilter: ident,
+	keyFilter: ident
+};
+
+/**
+ * Get an item from storage
+ *
+ * @function getItem
+ * @memberof Storage
+ * @instance
+ * @param {string} key - The key to get
+ * @return {*} - The value from storage
+ */
+Storage.prototype.getItem = function (key) {
+	return this.options.getValueFilter(this.backend.getItem(this.options.keyFilter(key)));
+};
+
+/**
+ * Sets an item in storage
+ *
+ * @function setItem
+ * @memberof Storage
+ * @instance
+ * @param {string} key - The storage key to set
+ * @param {*} value - The value to store
+ * @param {object} [params] - The optional storage params
+ */
+Storage.prototype.setItem = function (key, value, params = {}) {
+	params.domain = params.domain || this.options.domain;
+	return this.backend.setItem(this.options.keyFilter(key), this.options.setValueFilter(value), params);
+};
+
+/**
+ * Remove an item from storage
+ *
+ * @function removeItem
+ * @memberof Storage
+ * @instance
+ * @param {string} key - The key to remove
+ * @param {object} [params] - The params to use for removal
+ */
+Storage.prototype.removeItem = function (key, params = {}) {
+	params.domain = params.domain || this.options.domain;
+	return this.backend.removeItem(this.options.keyFilter(key), params);
+};
+
+/**
+ * Clear out an entire backend store
+ *
+ * @function clear
+ * @memberof Storage
+ * @instance
+ */
+Storage.prototype.clear = function () {
+	return this.backend.clear();
+};
+
+// Gets a particular storage backend by it's name
+Storage.backends = [storage('localStorage'), cookie, storage('sessionStorage'), memory];
+function getBackend (name) {
+	for (var i = 0, len = Storage.backends.length; i < len; i++) {
```

Comment:
oh well if you change this, you might as well also change the for loop a couple lines below this. didnt bother to comment because it is breaking early also which is likely technically better than [].filter